### ISSUE TITLE
InjectMongoClient within a constructor?

### ISSUE THREAD

**Issue by StephenOTT** [SoftInstigate/restheart#388]
Could the InjectMongoClient annotation support being used within a constructor of a service?

**Comment by ujibang** [comment]
no, the `@InjectionX` annotation can be used only on public methods, as in

```
@InjectMongoClient
public void init(MongoClient mclient) {
    this.mclient = mclient;
}
```

**Comment by StephenOTT** [comment]
@ujibang would injection as a constructor be possible given how you implement the injection?   I am looking at this from a kotlin perspective, and it would really cleanup the code to be able to add it into the constructor of a plugin
